sdks/python/apache_beam/runners/direct/direct_runner.py

```diff
@@ -86,6 +86,10 @@ def run(self, pipeline):
                         evaluation_context)
     # Start the executor. This is a non-blocking call, it will start the
     # execution in background threads and return.
+
+    from apache_beam.utils.value_provider import RuntimeValueProvider
+    if pipeline.options:
+      RuntimeValueProvider.set_runtime_options(pipeline.options._options_id, {})
```

Comment:
The options map didn't really grow so much for our current testing needs, but I have added a class method to clear the map